.travis.yml

```diff
@@ -46,11 +47,38 @@ jobs:
             repo: open-policy-agent/gatekeeper
             branch: master
             tags: false
-        - provider: script
-          skip_cleanup: true
-          script: make travis-dev-release
-          on:
-            repo: open-policy-agent/gatekeeper
-            branch: master
-            tags: true
-
+    - stage: "build, release"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request AND tag IS present
+      script:
+      - make manager
+      # Push new release version to registry
+      - make travis-dev-release IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG} VERSION=${TRAVIS_TAG}
+    - stage: "verify release"
+      name: "e2e on released image"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request AND tag IS present
+      before_script:
+      - KIND_VERSION=0.4.0
+      - KUSTOMIZE_VERSION=3.0.2
+      # Download and install kind
+      - curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output kind && chmod +x kind && sudo mv kind /usr/local/bin/
+      # Create kind cluster
+      - |
+        if [ $(kind get clusters) ]; then
+          export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+        else
+          kind create cluster
+        fi
+      # Get kubeconfig
+      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+      # Update manager_image_patch.yaml with latest release tag (git tag)
+      - make docker-build IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG}
```

Comment:
running docker-build is potentially counterproductive here, as it may prevent us from pulling the hosted image if there is a misconfiguration later in the pipeline. I suggest splitting manifest generation into a separate rule (looks like you do this with `release-deploy`, which I suggest renaming to be more descriptive). This will also improve runtime by avoiding a superfluous build.

Reply:
Agreed. Currently `docker-build` builds the image and generates the manifest match. I will break these out then.